import_flow_events.py

```diff
@@ -100,12 +101,15 @@
 """
 
 Q_CLEAR_DAY_METADATA = """
-    DELETE FROM flow_metadata
-    WHERE begin_time::DATE = '{day}'::DATE;
+    DELETE m
+    FROM flow_metadata AS m
+    LEFT JOIN flow_events AS e
+      ON m.flow_id = e.flow_id
+    WHERE e.flow_id IS NULL;
 """
 Q_CLEAR_DAY_EVENTS = """
     DELETE FROM flow_events
-    WHERE timestamp::DATE = '{day}'::DATE;
+    WHERE origin = '{day}'::DATE;
```

Comment:
One trick here is that we're no longer deleting by the sortkey, so this query will have to examine every item in the table, and hence will get more and more expensive over time.  Perhaps we could add additional `WHERE` clauses to restrict `timestamp` to be within 1 day either side of the target, so that it can avoid scanning rows that are definitely not going to match.